src/sys/socket/mod.rs

```diff
@@ -986,6 +987,7 @@ pub fn recvmsg<'a>(fd: RawFd, iov: &[IoVec<&mut [u8]>],
             sockaddr_storage_to_addr(&address, mhdr.msg_namelen as usize).ok()
         };
         RecvMsg {
+            bytes: ret as usize,
```

Comment:
While this will technically work, it's bad style because it won't work for other data types.  Instead of using `ret`, you should use the argument provided to `map`, like this:
```
Errno::result(ret).map(|r| {
    ...
    bytes: r as usize
}
```